### PR TITLE
log.c: warn instead of error on failed stdpath('cache') creation

### LINKED ISSUE #13771: Homebrew HEAD not compiling
- Operating system/version: macOS 11.1
- Terminal name/version: iTerm2 3.4.3
- `$TERM`: tmux-256color

### Steps to reproduce

```
brew install neovim --HEAD
```

I have tried suggestions for removing, unlinking, rebuilding luajit as suggested here: https://github.com/neovim/neovim/issues/13529

### Actual behaviour
```
❯ brew install neovim --HEAD
==> Downloading https://github.com/libmpack/libmpack-lua/releases/download/1.0.8/libmpack-lua-1.0.8.tar.gz
Already downloaded: /Users/david/Library/Caches/Homebrew/downloads/4b91349796a66e38e93acd77e6ca14212c869090429fd88271def2be5293d83f--libmpack-lua-1.0.8.tar.gz
==> Downloading https://luarocks.org/manifests/gvvaughan/lpeg-1.0.2-1.src.rock
Already downloaded: /Users/david/Library/Caches/Homebrew/downloads/0bc17975f0cf3981083172aa7da329b874e5780fa81494f5dd07d0665fcec23b--lpeg-1.0.2-1.src.rock
==> Downloading https://luarocks.org/manifests/kikito/inspect-3.1.1-0.src.rock
Already downloaded: /Users/david/Library/Caches/Homebrew/downloads/ecc1fa84ce4c83c9406f6d1202d4a88c80ede4a0a2bfd8d8d2bd54fd042226bb--inspect-3.1.1-0.src.rock
==> Downloading https://github.com/keplerproject/lua-compat-5.3/archive/v0.7.tar.gz
Already downloaded: /Users/david/Library/Caches/Homebrew/downloads/5f4e552e95b9a6223a5e4fabb7f04bf4b886057024507c76b1d5b803218c9363--lua-compat-5.3-0.7.tar.gz
==> Downloading https://github.com/luvit/luv/releases/download/1.30.0-0/luv-1.30.0-0.tar.gz
Already downloaded: /Users/david/Library/Caches/Homebrew/downloads/a9ce5a949370965d8f0fe70a47e7d102a3b4b6b831dbd4d088a83bbe1c00a3a2--luv-1.30.0-0.tar.gz
==> Cloning https://github.com/neovim/neovim.git
Updating /Users/david/Library/Caches/Homebrew/neovim--git
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at abbfaf286 logs: make kXDGCacheHome if it doesn't exist (#13758)
==> luarocks make --lua-dir=/usr/local/opt/luajit --tree=/private/tmp/neovim-20210116-4155-1txrl6u/deps-build
==> luarocks make --lua-dir=/usr/local/opt/luajit --tree=/private/tmp/neovim-20210116-4155-1txrl6u/deps-build
==> luarocks make --lua-dir=/usr/local/opt/luajit --tree=/private/tmp/neovim-20210116-4155-1txrl6u/deps-build
==> cmake . -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DCMAKE_INSTALL_PREFIX=/private/tmp/neovim-20210116-4155-1txrl6u/deps-bu
==> make install
==> cmake .. -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DLIBLUV_INCLUDE_DIR=/private/tmp/neovim-20210116-4155-1txrl6u/deps-bui
==> make install
Last 15 lines from /Users/david/Library/Logs/Homebrew/neovim/07.make:
cd /tmp/neovim-20210116-4155-1txrl6u/build/runtime && /usr/local/bin/luajit /tmp/neovim-20210116-4155-1txrl6u/scripts/genvimvim.lua /tmp/neovim-20210116-4155-1txrl6u/src/nvim /tmp/neovim-20210116-4155-1txrl6u/build/runtime/syntax/vim/generated.vim /tmp/neovim-20210116-4155-1txrl6u/build/funcs_data.mpack
cd /tmp/neovim-20210116-4155-1txrl6u/build/runtime/pack/dist/opt/vimball && /usr/local/Cellar/cmake/3.19.3/bin/cmake -E copy_directory /tmp/neovim-20210116-4155-1txrl6u/runtime/pack/dist/opt/vimball /tmp/neovim-20210116-4155-1txrl6u/build/runtime/pack/dist/opt/vimball
[100%] Generating pack/dist/opt/matchit/doc/tags
cd /tmp/neovim-20210116-4155-1txrl6u/build/runtime/pack/dist/opt/matchit && /usr/local/Cellar/cmake/3.19.3/bin/cmake -E copy_directory /tmp/neovim-20210116-4155-1txrl6u/runtime/pack/dist/opt/matchit /tmp/neovim-20210116-4155-1txrl6u/build/runtime/pack/dist/opt/matchit
cd /tmp/neovim-20210116-4155-1txrl6u/build/runtime && /usr/local/Cellar/cmake/3.19.3/bin/cmake -E copy_directory /tmp/neovim-20210116-4155-1txrl6u/runtime/doc doc
cd /tmp/neovim-20210116-4155-1txrl6u/build/runtime/pack/dist/opt/matchit && /tmp/neovim-20210116-4155-1txrl6u/build/bin/nvim -u NONE -i NONE -e --headless -c helptags\ doc -c quit
cd /tmp/neovim-20210116-4155-1txrl6u/build/runtime/pack/dist/opt/vimball && /tmp/neovim-20210116-4155-1txrl6u/build/bin/nvim -u NONE -i NONE -e --headless -c helptags\ doc -c quit
cd /tmp/neovim-20210116-4155-1txrl6u/build/runtime && /tmp/neovim-20210116-4155-1txrl6u/build/bin/nvim -u NONE -i NONE -e --headless -c helptags\ ++t\ doc -c quit
E5010: Failed to create directory /privateE/tmp/neovim5010: Failed to create directory /private/tmp/neovim-20210116-4-155-1txrl2021016u/.brew16-4155-1txrl6u/.brew_home/.cache for writing logs: file already exists_home/.cache for writing logs: file already existsmake[2]: *** [runtime/pack/dist/opt/vimball/doc/tags] Error 1
make[2]: *** Deleting file `runtime/pack/dist/opt/vimball/doc/tags'
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [runtime/doc/tags] Error 1
make[2]: *** Deleting file `runtime/doc/tags'
make[1]: *** [runtime/CMakeFiles/runtime.dir/all] Error 2
make: *** [all] Error 2

READ THIS: https://docs.brew.sh/Troubleshooting

Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels.
```

### Expected behaviour

It installs.

### PR DESCRIPTION
**Pull request by mjlbach**
Closes #13771

TLDR, brew and nix both use an immutable home directory in which we try to create a $HOME/.cache/nvim. We gracefully fail during runtime while reporting the error to the user (no one should be using an immutable cache at runtime), !however, reporting this error causes a problem during buildtime while running helptag generation. Which is why brew fails to install. The exact command is:
`/tmp/neovim-20210116-4155-1txrl6u/build/bin/nvim -u NONE -i NONE -e --headless -c helptags\ ++t\ doc -c quit`

You can reproduce the same issue at runtime by making ~/.cache/nvim not writeable by your user. It will report an error (in nvim) which can be dismissed.

### CODE REVIEW

File: src/nvim/log.c [hunk 1]
```diff
@@ -76,8 +76,8 @@ static bool log_path_init(void)
     if (!os_isdir((char_u *)cachehome)) {
         int ret;
         if ((ret = os_mkdir_recurse(cachehome, 0700, &failed_dir)) != 0) {
-          EMSG3(_(LOGERR "Failed to create directory %s "
-                  "for writing logs: %s"),
+          WLOG(LOGERR "Failed to create directory %s "
+                  "for writing logs: %s",
                 failed_dir, os_strerror(ret));
```
Comment (jamessan) [inline src/nvim/log.c:81]: Since it's not displayed to the user, there's no need for an error-code (so that can be removed from the docs, too).

Also, the formatting good be a little more compact, and lint wants `failed_dir` to line up with the column after the opening `(`.

```suggestion
          WLOG("Failed to create directory %s for writing logs: %s",
               failed_dir, os_strerror(ret));
```

File: src/nvim/log.c [hunk 2]
```diff
@@ -73,15 +72,10 @@ static bool log_path_init(void)
     // Make kXDGCacheHome if it does not exist.
     char *cachehome = get_xdg_home(kXDGCacheHome);
     char *failed_dir = NULL;
+    int log_dir_failure = 0;
```
Comment (jamessan) [inline src/nvim/log.c:75]:
```suggestion
    bool log_dir_failure = false;
```

File: src/nvim/log.c [hunk 3]
```diff
@@ -97,6 +91,12 @@ static bool log_path_init(void)
       return false;
     }
     os_setenv(LOG_FILE_ENV, log_file_path, true);
+    if (log_dir_failure != 0) {
```
Comment (jamessan) [inline src/nvim/log.c:94]:
```suggestion
    if (log_dir_failure) {
```

**Review by jamessan — APPROVED** [review]
